packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -881,8 +883,9 @@ class _AppBarState extends State<AppBar> {
           tooltip: MaterialLocalizations.of(context).openAppDrawerTooltip,
         );
       } else {
-        if (!hasEndDrawer && canPop)
+        if (canPop && !(hasEndDrawer && isFirstRoute && isEndDrawerOpen)) {
```

Comment:
```suggestion
final ____ !(hasEndDrawer && isFirstRoute && isEndDrawerOpen);
if (canPop && ____) {
```
Just to make it more clear what is the final purpose of the `!(hasEndDrawer && isFirstRoute && isEndDrawerOpen)`. Replace the `____` with a meaningful variable name.

Reply:
@xeinebiu I've replaced it. And also changed the parameter name that wasn't fit the style guide in the test file.